packages/core/src/rules/common/assertions/asserts.ts

```diff
@@ -134,4 +134,13 @@ export const asserts: Asserts = {
       ? getIntersectionLength(value, condition) === condition.length
       : true;
   },
+  ref: (_value: any, condition: string | boolean, rawValue: any): boolean => {
```

Comment:
I think this is not going to work correctly in terms it won't use `context` or `subeject` settings.

Try to add an end-to-end test when you have multiple references (PathItem and Parameter for example) but you want only some specific type of reference to be asserted.

Reply:
@And678 were you able to test it? Also #662 is likely to merge in the morning, and I don't know if it will impact your PR.

Reply:
@adamaltman Yes, managed to test it, fixing final issue, probably will release on monday morning

Reply:
Thanks for the update! 🎉

Reply:
@And678 I just double-checked the code. Sorry, I thought you were using `ref` visitor but this is a ref assertion 🤦 

Disregard my comment above.

Reply:
But, please, add e2e test. I can't see any. Thanks.

Reply:
I just tested it. It still doesn't work correctly even with the example from the PR description.

It always reports $ref to be missing (even if it's there). And if you try `ref: false` it never reports at all.

Please, add positive and negative e2e tests.